blog/2020-10-23-the-making-of-new-user-sdk-tutorials.md

```diff
@@ -0,0 +1,110 @@
+
+A tool that automates the synchronization of code with the docs from any given repository has several key benefits:
+
+1. Code snippets in the documentation are guaranteed to work as they are continuously tested. This also means that they can be reliably copied and pasted into the user's editor.
+2. We control exactly which lines of code are shown and can also target a specific branch or commit. This is a great safeguard against bugs which might be introduced to the main branch.
+3. We never have to commit source code into the docs. The code is merged into the Markdown at build time. This ensures that the code is already reviewed and approved from the repo it resides in.
+
+Snipsync does come with a few considerations:
+
+1. Embedded code needs to have carefully reviewed comments, structure, and make sense within the context of the documentation. For example, if the code snippet is coming from a working repo it may include additional variables or function calls. These must be minimized and optimized out so they don’t cause unnecessary confusion.
+2. In the same way that the code must be optimized for the docs, the docs must also be optimized for the code. In essence, the docs are being “driven” and "defined" by the underlying code. And if no one has coined the term yet, I think the credit for “code-driven documentation” should go to our Head of Product, Ryland Goldstein as he pinged me early one morning to share that epiphany with me.
+
+We decided to embrace [Snipsync](https://github.com/temporalio/snipsync) and solve the challenges that it introduced.
+
+## Template repos
+
+<img alt="image" class="docs-image-centered docs-image-max-width-20" src="https://raw.githubusercontent.com/temporalio/documentation-images/main/static/templates.png" />
+
+We now had a way to synchronize code with our documentation. But from where will the code be synchronized? We know users will likely want to view the source file and relative file path of the code snippet for added context. They will also be likely to clone the repo and try to run the sample.
+
+We actually already repositories of code samples for the [Go SDK](https://github.com/temporalio/samples-go) and [Java SDK](https://github.com/temporalio/samples-java). And while we desired to see more samples, there were already quite a few of them in each. But, we discovered that shared sample repositories tended to have two issues that made them less ideal for synchronizing with docs.
```

Comment:
Typo in the first sentence, but also I really don't recommend starting a new para with "actually'. It makes the reader feel out of the loop.

Reply:
```suggestion
We actually already repositories of code samples for the [Go SDK](https://github.com/temporalio/samples-go) and [Java SDK](https://github.com/temporalio/samples-java). While we desired to see more samples, there were already quite a few of them in each repository. But, we discovered that shared sample repositories tend to have two issues that made them less ideal for synchronizing with docs.
```